lib/ansible/modules/windows/win_domain_computer.ps1

```diff
@@ -0,0 +1,183 @@
+      -DNSHostName $desired_state.dns_hostname `
+      -Enabled $desired_state.enabled `
+      -Description $desired_state.description `
+      -WhatIf:$check_mode
+  } Catch { Fail-Json $result "Error in function Set-ConstructedState" }
+
+  If ($initial_state.distinguished_name -ne $desired_state.distinguished_name)
+  {
+    # Move computer to OU
+    Try {
+      Get-ADComputer $desired_state.name |
+          Move-ADObject `
+            -TargetPath $desired_state.ou `
+            -Confirm:$False `
+            -WhatIf:$check_mode
+    } Catch { Fail-Json $result "Error in function Move-ADObject" }
+  }
+  Set-Attr $result "changed" $true
+}
+
+# ------------------------------------------------------------------------------
+Function Add-ConstructedState($desired_state) {
+  Try {
+    New-ADComputer `
+      -Name $desired_state.name `
+      -SamAccountName $desired_state.sam_account_name `
+      -DNSHostName $desired_state.dns_hostname `
+      -Path $desired_state.ou `
+      -Enabled $desired_state.enabled `
+      -Description $desired_state.description `
+      -WhatIf:$check_mode
+  } Catch { Fail-Json $result "Error in function Add-ConstructedState" }
+
+  Set-Attr $result "changed" $true
+}
+
+# ------------------------------------------------------------------------------
+Function Remove-ConstructedState($initial_state) {
+  Try {
+    Remove-ADComputer `
+      -Identity $initial_state.name `
+      -Confirm:$False `
+      -WhatIf:$check_mode
+  } Catch { Fail-Json $result "Error in function Remove-ADComputer" }
+
+  Set-Attr $result "changed" $true
+}
+
+# ------------------------------------------------------------------------------
+Function are_hashtables_equal($x, $y) {
+  # Compare not nested HashTables
+  Foreach ($key in $x.Keys) {
+      If (($y.Keys -notcontains $key) -or ($x[$key] -ne $y[$key])) {
+          Return $false
+      }
+  }
+  foreach ($key in $y.Keys) {
+      if (($x.Keys -notcontains $key) -or ($x[$key] -ne $y[$key])) {
+          Return $false
+      }
+  }
+  Return $true
+}
+
+# ------------------------------------------------------------------------------
+$desired_state = Get-DesiredState($params)
+$initial_state = Get-InitialState($desired_state)
+
+If ($desired_state.state -eq "Present") {
+    If ($initial_state.state -eq "Present") {
+      $in_desired_state = are_hashtables_equal $initial_state $desired_state
+
+      If (-not $in_desired_state) {
+        Set-ConstructedState $initial_state $desired_state
+      }
+    } Else { # $desired_state.state = "Present" & $initial_state.state = "Absent"
+      Add-ConstructedState($desired_state)
+    }
+  } Else { # $desired_state.state = "Absent"
+    If ($initial_state.state -eq "Present") {
+      Remove-ConstructedState($initial_state)
+    }
+  }
+
+If ($diff_support) {
+  $diff = @{
+    before = $initial_state
+    after = $desired_state
+  }
+  Set-Attr $result "diff" $diff
+}
+
+Exit-Json $result
```

Comment:
Try to avoid omitted parameters with cmdlets, it makes it a lot easier for people who don't know much about it what each value is for, e.g. `Exit-Json -obj $result`

Reply:
Is there any code formatter for powershell? Anything that:
* Adds default values operators.
* Resolves alias to canonical form
* Converts commands to its intended case: exit-json => Exit-Json
* Indents the code

(or a subset of it).

Reply:
I use visual studio code, it gives you the things you mentioned and a pretty good debugger as well. The Legacy.psm1 files are loaded in the intellisence once it has been imported once.